### ISSUE TITLE
getAdapterPosition() can be wrong

### ISSUE THREAD

**Issue by scriptacus** [iPaulPro/Android-ItemTouchHelper-Demo#15]
This isn't a bug with this code, but I'm throwing it out there as a potential gotcha that I only ran into after implementing the dismiss/swap functionality from this demo.

After a dismiss or swap of items in a recycler view, "holder.getAdapterPosition()" will return NO_POSITION (-1) if you've called "notifyDataSetChanged()" since the last layout pass.  I was passing getAdapterPosition() to my Activity in response to some user interaction, which had always worked previously.  After either of the actions above, the adapter position became invalid until an orientation change or attempt to scroll the screen.  Using getLayoutPosition() in place of getAdapterPosition() resolved the issue.

**Comment by Farasy** [comment]
Thanks for info.

**Comment by pixelbendr** [comment]
Yes I noticed this, my app start crashes sometimes on swipe; when I run the test getAdapterPosition is -1. :( is there a fix for this.

**Comment by noureldeen-abouelkassem** [comment]
it is better to use notifyItemInserted(0) or notifyItemRemoved(0) cause it keeps the calculations of the adapterPosition which doesn't happen when you use notifyDataSetChanged() cause when this happens you redraw a new layout and restructure the position of the adapter .
also you can avoid this problem by passing your recycler view to your adapter and use getLayoutPosition () mRecyclerView.findViewHolderForLayoutPosition(myViewHolder.getLayoutPosition());

hope this help
